### ISSUE TITLE
Some animations are stuttering, hack helps

### ISSUE THREAD

**Issue by bobbelkl** [airbnb/lottie-ios#1296]
I have a large animation that is lagging on playback. I was debugging for hours an found a workaround with CADisplayLink.
Checking the realtimeAnimationProgress every few milliseconds makes the animation smooth again.

And that’s the hack I'm using.

```
override func viewDidLoad() {
        super.viewDidLoad()
        displayLink = CADisplayLink(target: self, selector: #selector(smoothAnimationHack))
        displayLink?.add(to: .current,
                        forMode: RunLoop.Mode.default)
}
    
@objc func smoothAnimationHack() {
        _ = animationView.realtimeAnimationProgress
}
```

I tried it with different devices/simulators, but it's same the behavior everywhere.
Please don't use the example app or the LottieFiles App to check, both use CADisplayLink to update the slider.

## Check these before submitting:
- [x] The issue doesn't involve an [Unsupported Feature](https://github.com/airbnb/lottie-ios/blob/master/README.md#supported-after-effects-features)
- [x] This issue isn't related to another open issue

## This issue is a:
- [x] Non-Crashing Bug (Visual or otherwise)
- [] Crashing Bug
- [] Feature Request
- [] Regression (Something that once worked, but doesn't work anymore)

## Which Version of Lottie are you using?

Lottie 3.1.9

## What Platform are you on? 

- [] MacOS
- [x] iOS

## What Language are you in?
- [x] Swift
- [] Objective-C

## Animation JSON

Here is my animation:
https://assets8.lottiefiles.com/packages/lf20_e0selqdr.json

**Comment by kapsyst** [comment]
I have experienced similar behavior playing an animation while running a SceneKit rendering thread. Updating an unrelated view on the main thread as below alleviates the problem, but I want to remove this call anyway, so It would be nice if there was an alternate solution.
```
// SceneKit delegate callback
public func renderer(_ renderer: SCNSceneRenderer, updateAtTime time: TimeInterval) {
    DispatchQueue.main.async {
        self.myView.updateMyViewWidget(CGFloat(0))
    }
}
```
My animation is also very large (> 2000 frames), and the initial play call (to a loop) is made on the main thread. Any help appreciated.

**Comment by kapsyst** [comment]
Some more updates:
I'm also updating a camera output layer via AR Augmented Faces. If I disable this and leave the render thread empty, the animation is smooth as normal. Adding only the camera output, as below, causes the stuttering.
```
    public func renderer1(_ renderer: SCNSceneRenderer, updateAtTime time: TimeInterval) {

        guard let frame = self.faceSession?.currentFrame else {
            return
        }

        let imageBuffer = frame.capturedImage

        if CVPixelBufferLockBaseAddress(imageBuffer, .readOnly) == kCVReturnSuccess {
            if CVPixelBufferIsPlanar(imageBuffer) {
                self.cameraImageLayer.contents = imageBuffer
            }
        }

        if CVPixelBufferUnlockBaseAddress(imageBuffer, .readOnly) != kCVReturnSuccess {
            fatalError()
        }
    }
```
Adding the following code to the end of the function fixes the problem (has to be run from main thread).
```
        DispatchQueue.main.async {
            _ = self.animationView.realtimeAnimationProgress
        }
```

**Comment by reggian** [comment]
Thank you very much for this hack.

**Comment by calda** [comment]
Today we released Lottie 3.4.0 which includes a new rendering engine powered by Core Animation ([more details here](https://github.com/airbnb/lottie-ios/discussions/1627)). This new rendering engine has significantly better performance, and should resolve the performance issues folks have been encountering with the previous main thread rendering engine. Please try it out by setting the `LottieConfiguration.renderingEngine` feature flag, and feel free to report any new issues that you experience when using the new rendering engine.
